### PR TITLE
feat: implement schema description and validation

### LINKED ISSUE #606: Describe tdp vars
We need a way to provide a comprehensive description of TDP Variables.

JSON Schema might be a good candidate.

### PR DESCRIPTION
**Pull request by gboutry**
#### Which issue(s) this PR fixes

Fixes #606

#### Additional comments

This PR bootstraps schema description and validation implementation.

#### Agreements

- [X] I hereby declare this contribution to be licensed under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0).
- [X] I hereby agree to grant [TOSIT](https://www.tosit.io/) a copyright license to use my contributions.

### CODE REVIEW

File: tdp_vars_schema/hadoop.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "$id": "TOSIT-IO/schema/hadoop.json",
```
Comment (rpignolet) [inline tdp_vars_schema/hadoop.json:3]: Maybe add `TDP` in the `id`, the TOSIT can have other projects.

Comment (gboutry) [reply tdp_vars_schema/hadoop.json]: Done

**Comment by leopaul36** [comment]
Looks good to me!

If I understand correctly, with this approach:
- `tdp_vars_defaults` will be deprecated in favor of their equivalent as `tdp_vars_schema` where we will define all variables with a type, description, default value, etc.
- `tdp init` will generate `inventory/tdp_vars/*/*.yml` files by taking every properties of the JSON schemas and initiating them with their default values
- `validate_schema` should be run before each deployment in `tdp-lib` against the `TDP_VARS` dir

Do I get something wrong?

**Comment by gboutry** [comment]
> Looks good to me!
> 
> If I understand correctly, with this approach:
> 
> - `tdp_vars_defaults` will be deprecated in favor of their equivalent as `tdp_vars_schema` where we will define all variables with a type, description, default value, etc.
That's not the case. For the moment, defaults are still defined in `tdp_vars_defaults`
> - `tdp init` will generate `inventory/tdp_vars/*/*.yml` files by taking every properties of the JSON schemas and initiating them with their default values

No

> - `validate_schema` should be run before each deployment in `tdp-lib` against the `TDP_VARS` dir

Well, it's a comment for tdp-lib https://github.com/TOSIT-IO/tdp-lib/pull/290, I did not implement schema validation on deploy, there's only on `init` command, and on demand with `validate`
> 
> Do I get something wrong?

**Review by rpignolet — APPROVED** [review]

**Comment by gboutry** [comment]
is it ok with you @leopaul36 ?

**Pull request merged**